src/projection.rs

```diff
@@ -117,6 +117,11 @@ impl<S: BaseFloat> From<PerspectiveFov<S>> for Matrix4<S> {
             "The vertical field of view cannot be greater than a half turn, found: {:?}",
             persp.fovy
         );
+        assert!(
+            persp.aspect.abs() > S::zero(),
```

Comment:
can we use the approx methods/macros here, like https://docs.rs/approx/0.3.2/approx/macro.assert_abs_diff_ne.html